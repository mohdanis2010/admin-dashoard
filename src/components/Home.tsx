import * as React from "react";
import { RouteComponentProps } from "react-router-dom";
import axios from "axios";
import PaginationFilter from "./PaginationFilter";

interface IState {
  customers: any[];
}

export default class Home extends React.Component<RouteComponentProps, IState> {
  constructor(props: RouteComponentProps) {
    super(props);
    this.state = { customers: [] };
  }

  public componentDidMount(): void {
    axios.get(`http://localhost:5000/customers`).then((data) => {
      this.setState({ customers: data.data });
    });
  }

  public deleteCustomer = (id: number) => {
    axios.delete(`http://localhost:5000/customers/${id}`).then((data) => {
      const index =
        this.state &&
        this.state.customers.findIndex((customer) => customer.id === id);
      this.state.customers.splice(index, 1);
      this.props.history.push("/");
    });
  };

  public render() {
    const customers = this.state.customers;

    return (
      <div>
        {customers.length === 0 && (
          <div className="text-center">
            <h2>No customer found at the moment</h2>
          </div>
        )}

        <div className="container">
          <div className="row">
            <PaginationFilter
              customers={customers}
              deleteCustomer={this.deleteCustomer}
            />
          </div>
        </div>
      </div>
    );
  }
}
