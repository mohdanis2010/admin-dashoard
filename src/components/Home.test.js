import React from "react";
import { render, cleanup } from '@testing-library/react';
import Home from './Home';

afterEach(cleanup);

it("renders", () => {
  const { asFragment } = render(<Home customers={[]} />);
  expect(asFragment()).toMatchSnapshot();
});