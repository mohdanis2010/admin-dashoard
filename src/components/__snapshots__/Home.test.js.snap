// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div>
    <div
      class="text-center"
    >
      <h2>
        No customer found at the moment
      </h2>
    </div>
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="sc-AxjAm jaCnSd"
        >
          <label
            class="search-text"
          >
            Search Customer:
            <input
              name="search"
              type="search"
              value=""
            />
          </label>
          <div
            class="sc-AxheI emqWSV"
          >
            No matching customer.
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
